barretenberg/cpp/src/barretenberg/vm2/common/standard_affine_point.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <cstdint>
+
+namespace bb::avm2 {
+
+template <typename AffinePoint> class StandardAffinePoint {
+  public:
+    using BaseField = AffinePoint::Fq;
+    using ScalarField = AffinePoint::Fr;
+
+    constexpr StandardAffinePoint() noexcept
+        : point(AffinePoint())
+    {}
+
+    constexpr StandardAffinePoint(AffinePoint val) noexcept
+        : point(val)
+    {}
+
+    constexpr StandardAffinePoint(BaseField x, BaseField y, bool is_infinity) noexcept
+    {
+        if (is_infinity) {
+            point = AffinePoint::infinity();
+        } else {
+            point = AffinePoint(x, y);
+        }
+    }
+
+    constexpr StandardAffinePoint operator+(const StandardAffinePoint& other) const noexcept
+    {
+        return StandardAffinePoint(point + other.point);
+    }
+
+    constexpr StandardAffinePoint operator*(const ScalarField& exponent) const noexcept
+    {
+        return StandardAffinePoint(point * exponent);
+    }
+
+    constexpr bool operator==(const StandardAffinePoint& other) const noexcept { return point == other.point; }
+
+    constexpr StandardAffinePoint operator-() const noexcept { return StandardAffinePoint(-point); }
+
+    [[nodiscard]] constexpr bool is_infinity() const noexcept { return point.is_point_at_infinity(); }
+
+    [[nodiscard]] constexpr bool on_curve() const noexcept { return point.on_curve(); }
+
+    constexpr BaseField x() const noexcept { return point.is_point_at_infinity() ? BaseField::zero() : point.x; }
```

Comment:
you probably should make this a `const BaseField&` so that the copy is optional. Same below.